modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -92,6 +92,29 @@ function toggleFilesInCart(incomingFile) {
   };
 }
 
+function removeFilesFromCart(files) {
+  return (dispatch, getState) => {
+    const filesToRemove = _.isArray(files) ? files : [files];
+    const existingFiles = getState().cart.files;
+    const nextFiles = _.differenceBy(existingFiles, filesToRemove, 'file_id');
+    const filesRemoved = existingFiles.length - nextFiles.length;
+    dispatch(notify(getNotificationComponent('remove', `remove/${filesToRemove.file_name}`, {
```

Comment:
Wouldn't `filesToRemove.file_name` be undefined if its an array? 
Also, I noticed the `id` we pass to notification has to be unique. In this case if you remove files once wait for notification to go away and remove again, the new notification won't show up. This also happens with Files table. Somehow we need to generate unique IDs each time. Do you mind taking a look? Or should we create a different ticket for this as its a bug with Files table as well?